server/routes/induction/common/previousWorkExperienceDetailController.ts

```diff
@@ -0,0 +1,75 @@
+import createError from 'http-errors'
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import type { PreviousWorkExperienceDetailForm } from 'inductionForms'
+import type { InductionDto } from 'inductionDto'
+import InductionController from './inductionController'
+import PreviousWorkExperienceDetailView from './previousWorkExperienceDetailView'
+import TypeOfWorkExperienceValue from '../../../enums/typeOfWorkExperienceValue'
+import logger from '../../../../logger'
+
+/**
+ * Abstract controller class defining functionality common to both the Create and Update Induction journeys.
+ */
+export default abstract class PreviousWorkExperienceDetailController extends InductionController {
+  constructor() {
+    super()
+  }
+
+  /**
+   * Returns the Previous Work Experience Detail view; suitable for use by the Create and Update journeys.
+   */
+  getPreviousWorkExperienceDetailView: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ): Promise<void> => {
+    const { prisonerSummary, inductionDto } = req.session
+    const { typeOfWorkExperience } = req.params
+
+    const previousWorkExperienceType = Object.values(TypeOfWorkExperienceValue).find(
+      type => type === typeOfWorkExperience.toUpperCase(),
+    )
+    if (!previousWorkExperienceType) {
```

Comment:
There is a little more complexity in this controller than the others that we are doing, in that this one has the type of job type on the request path: eg:
```
/prisoners/A1234BC/induction/previous-work-experience/retail`
```
The last element in the path needs to:
1. be a valid job type from the enum `TypeOfWorkExperienceValue`
2. be present on the prisoner's Induction

If either checks fail we return a 404

Reply:
Not tempted with a query param?